src/Parser/DeathKnight/Frost/Modules/Abilities.js

```diff
@@ -107,100 +159,123 @@ class Abilities extends CoreAbilities {
         },
         enabled: combatant.hasTalent(SPELLS.GLACIAL_ADVANCE_TALENT.id),
       },
+      // DEFENSIVE
       {
-        spell: SPELLS.BLINDING_SLEET_TALENT,
-        category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        spell: SPELLS.ANTI_MAGIC_SHELL,
+        buffSpellId: SPELLS.ANTI_MAGIC_SHELL.id,
+        category: Abilities.SPELL_CATEGORIES.DEFENSIVE,
+        gcd: null,
         cooldown: 60,
+        isDefensive: true,
+      },
+      {
+        spell: SPELLS.ICEBOUND_FORTITUDE,
+        buffSpellId: SPELLS.ICEBOUND_FORTITUDE.id,
+        category: Abilities.SPELL_CATEGORIES.DEFENSIVE,
+        gcd: null,
+        cooldown: 180,
+        isDefensive: true,
+      },
+      {
+        spell: SPELLS.DEATH_STRIKE,
+        category: Abilities.SPELL_CATEGORIES.DEFENSIVE,
         gcd: {
           base: 1500,
         },
-        enabled: combatant.hasTalent(SPELLS.BLINDING_SLEET_TALENT.id),
+        isDefensive: true,
       },
       {
-        spell: SPELLS.HORN_OF_WINTER_TALENT,
-        category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        spell: SPELLS.DEATH_PACT_TALENT,
+        buffSpellId: SPELLS.DEATH_PACT_TALENT.id,
+        category: Abilities.SPELL_CATEGORIES.DEFENSIVE,
+        gcd: null,
+        cooldown: 120,
+        isDefensive: true,
+        enabled: combatant.hasTalent(SPELLS.DEATH_PACT_TALENT.id),
+      },
+      // UTILITY
+      {
+        spell: SPELLS.DEATH_GRIP,
+        category: Abilities.SPELL_CATEGORIES.UTILITY,
         gcd: {
-          base: 1500,
-        },
-        cooldown: 45,
-        enabled: combatant.hasTalent(SPELLS.HORN_OF_WINTER_TALENT.id),
-        castEfficiency: {
-          suggestion: true,
-          recommendedEfficiency: 0.90,
+          static: 500,
         },
+        cooldown: 25,
       },
       {
         spell: SPELLS.CHAINS_OF_ICE,
-        category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        category: Abilities.SPELL_CATEGORIES.ROTATIONAL, //special case, this is rotational for the duration of prepatch
```

Comment:
Is this only with the chest legendary? In which case you can create a check if it's equipped I assume

Reply:
There's a talent that does the same thing so now that I think about it a permanent check for the talent would probably be the best